packages/react-components/src/components/chart/baseChart.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { useECharts } from '../../hooks/useECharts';
+import { ChartOptions } from './types';
+import { useVisualizedDataStreams } from './useVisualizedDataStreams';
+import { convertOptions } from './converters/convertOptions';
+
+/**
+ * Base chart to display Line, Scatter, and Bar charts.
+ */
+const Chart = ({ viewport, queries, size, ...options }: ChartOptions) => {
+  const { hasError, isLoading, dataStreams } = useVisualizedDataStreams(queries, viewport);
+
+  const option = convertOptions(
```

Comment:
we need to keep an "global object" per chart for option.. https://echarts.apache.org/en/api.html#echartsInstance.getOption.. we cannot lose the object given we will it to add or delete elements